Feature/add social media and location to organisation

appsol opened the pull request:
- Added Social Media input component to organisation form
- Tested Add / Delete Social Media to organisation
- Added Location select or add new form to organisation form
- Tested Add / Update existing location to organisation
- Tested Add new location to organisation
- Updated node-sass to 4.14 as current version was incompatible with node version. 4.14 is backward compatible so should present no issues.
- Ran lint --fix. This appears to have altered many files. Apologies for that.

matthew-inamdar commented on src/views/organisations/Edit.vue:
```diff
@@ -12,7 +12,7 @@
             <gov-body>General details about the organisation. To be found when searching or linked from a service page.</gov-body>
 
-            <organisation-form
+            <organisation-form v-if="!loading"
```
This isn't needed as it's wrapped in a `<template>` which already has this condition.

```suggestion
            <organisation-form
```

appsol replied:
Done

matthew-inamdar commented on src/views/organisations/Edit.vue:
```diff
@@ -66,9 +68,11 @@ export default {
+        email: this.organisation.email || "",
+        phone: this.organisation.phone || "",
+        social_medias: this.organisation.social_medias,
+        location_id: this.organisation.location.id,
```
Should be `location_id`.

```suggestion
        location_id: this.organisation.location_id,
```

appsol replied:
No this is correct. The location property of the returned organisation object is a full location object, not an ID reference.

matthew-inamdar replied:
What would happen if we try to edit an organisation that doesn't already have an associated location? Would we not get a `trying to access property of null` error?

appsol replied:
Fixed. Ternary operator checks for existence of organisation.location.

matthew-inamdar commented on src/views/organisations/Edit.vue:
```diff
@@ -91,10 +95,10 @@ export default {
           if (data.url === this.organisation.url) {
             delete data.url;
           }
```
Support now nullable value.

```suggestion
          if (data.url === (this.organisation.url || "")) {
            delete data.url;
          }
```

appsol replied:
Done

matthew-inamdar commented on src/views/organisations/Edit.vue:
```diff
@@ -91,10 +95,10 @@ export default {
             delete data.url;
           }
-          if (data.email === (this.organisation.email || '-')) {
+          if (data.email === (this.organisation.email || "-")) {
             delete data.email;
           }
```
Will be an empty string if the input is empty.

```suggestion
          if (data.email === (this.organisation.email || "")) {
            delete data.email;
          }
```

appsol replied:
Done

matthew-inamdar commented on src/views/organisations/Edit.vue:
```diff
@@ -91,10 +95,10 @@ export default {
             delete data.email;
           }
-          if (data.phone === (this.organisation.phone || '-')) {
+          if (data.phone === (this.organisation.phone || "-")) {
             delete data.phone;
           }
```
Will be an empty string if the input is empty.

```suggestion
          if (data.phone === (this.organisation.phone || "")) {
            delete data.phone;
          }
```

appsol replied:
Done

matthew-inamdar reviewed (changes requested):
Looking good 👍 

I've left a few code changes to take a look at as part of the review. I've also left some notes below:

When creating a new organisation, although the location and social media inputs are displayed, they are never sent to the API as part of the request.

The "view organisation page" doesn't show the social medias or location. Not sure if this was a requirement though? I would have assumed it was?

When editing an existing organisation, the location input is always empty, even if there is a location associated.

On the location input, when selecting an existing location, it is always in an error state (vertical red bar to the left of input).

When updating an organisation, it always provides the social medias, even when they remain unchanged. We need to check if they're unchanged and remove them from the request. Take a look at how we achieved this for services: 
https://github.com/Healthy-London-Partnership/admin/blob/38255f5ce8adca78a1e2ed32c0aa23934a7f6ffa/src/views/services/Edit.vue#L415-L420

Feel free to give me a shout if you want to go through any of it. I know the admin app is difficult to navigate through.

appsol commented:
- Fixed the create view to send the location_id and the social_medias.
- Added support for location and social media to the show view.
- Added filters to the edit view submission to remove location_id and social_medias if unchanged
- Fixed the location select error state. I cannot see an occasion when this would be in an error state so have defined the error prop as null.
- When an organisation is edited, the location is shown in the existing drop down. The location input should be empty as this is to add a new location not to display the current.

matthew-inamdar commented on src/views/organisations/Edit.vue:
```diff
@@ -88,16 +92,27 @@ export default {
             delete data.description;
           }
-          if (data.url === this.organisation.url) {
+          if (data.url === this.organisation.url || '') {
```
We need to wrap the logical "or" operator in parenthesis, otherwise, JS interprets this as:
```js
if ((data.url === this.organisation.url) || '') { /*...*/ }
```

```suggestion
          if (data.url === (this.organisation.url || '')) {
```

appsol replied:
Done. Sorry stupid mistake.

matthew-inamdar commented on src/views/organisations/Edit.vue:
```diff
@@ -88,16 +92,27 @@ export default {
             delete data.phone;
           }
 
+          if (data.location_id === this.organisation.location_id) {
+            delete data.location_id;
+          }
```
There's an issue here linked to this comment https://github.com/Healthy-London-Partnership/admin/pull/8/files#r489333468

This always resolves to `false` since there isn't a `location_id` on the `organisation` object.

matthew-inamdar reviewed (changes requested):
Seems like this is almost ready 👍 

Some slight issues with the `location_id` to take a look at.
